crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,208 @@
+use clap::Parser;
+use notify::{self, RecursiveMode, Watcher};
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use tokio::sync::mpsc;
+use tokio::sync::Mutex;
+use tokio::time;
+
+use crate::commands::build;
+
+use super::build::clients::LoamEnv;
+
+struct RebuildState {
+    pending: bool,
+}
+
+enum Message {
+    FileChanged,
+}
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub build_cmd: build::Cmd,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Watcher(#[from] notify::Error),
+    #[error(transparent)]
+    Build(#[from] build::Error),
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+fn canonicalize_path(path: &Path) -> PathBuf {
+    if path.as_os_str().is_empty() {
+        env::current_dir().unwrap_or_else(|_| PathBuf::from("."))
+    } else {
+        fs::canonicalize(path).unwrap_or_else(|_| path.to_path_buf())
+    }
+}
+
+fn is_parent_in_watched_dirs(parent: &Path, watched_dirs: &[Arc<PathBuf>]) -> bool {
+    watched_dirs.iter().any(|p| canonicalize_path(p) == parent)
+}
+
+fn is_temporary_file(path: &Path) -> bool {
+    let file_name = path.file_name().and_then(|s| s.to_str()).unwrap_or("");
+
+    // Vim temporary files
+    if file_name.starts_with('.') {
+        return true;
+    }
+    if file_name.ends_with('~') {
+        return true;
+    }
+
+    // Emacs temporary files
+    if file_name.starts_with('#') && file_name.ends_with('#') {
+        return true;
+    }
+    // VSCode temporary files
+    if std::path::Path::new(file_name)
+        .extension()
+        .map_or(false, |ext| ext.eq_ignore_ascii_case("tmp"))
+    {
+        return true;
+    }
+
+    // Add more patterns for other editors as needed
+
+    false
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<(), Error> {
```

Comment:
```suggestion
    pub async fn run(&mut self) -> Result<(), Error> {
```

Reply:
We can do this because the caller has a mutable reference! Then you can do.
```
         self.build_cmd
            .build_clients
            .env
            .get_or_insert(LoamEnv::Development);
        self.build_cmd.profile.get_or_insert_with(|| "debug".to_string());
        let build_cmd = Arc::new(build_cmd.clone());
```

Then below you can replace `self.clone()` with `build_cmd.clone()`, which just increments the counter and creates a new reference.

Reply:
Okay, refactored to only clone build_cmd. Let me know what you think.

Reply:
I reverted the changes because it was causing a failure during the build command. I'm not sure why what I did didn't work. If you think its important I can make another attempt at it.